src/common/gui/option.cpp

```diff
@@ -141,27 +142,52 @@ OptionBox::OptionBox() :
 
 void OptionBox::Update(uint32_t frame_time)
 {
-    _event = 0; // Clear all events
+    // Clear all of the events.
+    _event = 0;
 
-    if(!_scrolling)
+    if (!_scrolling) {
         return;
+    }
+
+    _scroll_time += frame_time;
+    if (_scroll_time > VIDEO_OPTION_SCROLL_TIME) {
+        _scroll_time = VIDEO_OPTION_SCROLL_TIME;
+    }
+
+    // Computes the scroll offset independently from the coordinate system.
+    _scroll_offset = static_cast<int32_t>((static_cast<float>(_scroll_time) / static_cast<float>(VIDEO_OPTION_SCROLL_TIME)) * _cell_height);
+
+    assert(_scroll_direction != 0);
+    if (_scroll_direction < 0) {
+        // Scroll up.
+        _scroll_offset = _cell_height - _scroll_offset;
+    }
 
-    if(!_scrolling_animated || _scroll_time > VIDEO_OPTION_SCROLL_TIME) {
+    if (!_scrolling_animated || _scroll_time >= VIDEO_OPTION_SCROLL_TIME) {
         _scroll_time = 0;
         _scrolling = false;
         _scroll_offset = 0.0f;
-        return;
-    }
 
-    _scroll_time += frame_time;
+        if (_scrolling_horizontally) {
```

Comment:
During an animation, we need to display more elements than normal.  For example, if a menu shows five elements at time, while animating we need to show six to properly display the new element which is animating into place.

This code updates the display variables when an animation completes.